File: Study/EditText and Button/MainActivity.java
package com.example.edittextex;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    EditText et_name; //단순히 버튼을 가리키는 레퍼런스 변수
    Button btn_submit;

    @Override
    protected void onCreate(Bundle savedInstanceState) { //앱 실행시 처음으로 동작되는 실행주기
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        et_name = findViewById(R.id.et_name); //실제 xml에 있는 EditText를 레퍼런스 변수에 연결한 것.
        btn_submit = findViewById(R.id.btn_submit);

        btn_submit.setOnClickListener(new View.OnClickListener() { //익명클래스로 바로 리스너 생성 후 연결함.
            @Override
            public void onClick(View v) { //버튼에 클릭 이벤트 발생시 처리할 문장.
                et_name.setText("다시 입력해주세요...");
            }
        });
    }
}

File: Study(Kotlin)/Intent/StartActivityForResult/MainActivity.kt
package com.example.intentsample

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import androidx.activity.result.ActivityResult
import androidx.activity.result.contract.ActivityResultContracts

const val EXTRA_MESSAGE = "com.example.intentsample.EXTRA_MESSAGE"
const val RETURN_MESSAGE = "com.example.intentsample.RETURN_MESSAGE"

class MainActivity : AppCompatActivity() {

    lateinit var editText: EditText;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button: Button = findViewById(R.id.button)
        editText = findViewById(R.id.editText)

        button.setOnClickListener {
            var str = editText.text.toString()
            var subIntent = Intent(this, SubActivity::class.java)
            subIntent.putExtra(EXTRA_MESSAGE,str)
            startForResult.launch((subIntent)) //결과를 돌려받기 위한 Activity 호출!
        }
    }

    private val startForResult = registerForActivityResult( // 결과를 돌려받았을 때 처리할 메소드.
        ActivityResultContracts.StartActivityForResult() )
    { result: ActivityResult ->
        if(result.resultCode == Activity.RESULT_OK)
            if(result.data != null){ // result.data는 intent 객체를 가리킴.
                val extras:Bundle? = result.data?.extras
                val returnString = extras?.getString(RETURN_MESSAGE) ?: ""
                editText.setText(returnString)
            }
    }
}
